File: Android License/Java/FPS/app/src/main/java/com/example/fps/MainActivity.java
/*
 * Title: MainActivity file.
 * Author: Terrortronics / Bradley Elenbaas (mr.elenbaas@gmail.com)
 * Version: 2
 * Date: September 15, 2023
 *
 * Intellectual Property:
 * Copyright (c) 2023 Bradley Elenbaas. All rights reserved.
 *
 * License:
 * This file is owned by Terrortronics / Bradley Elenbaas.
 * This file observes the Android License.
 */

package com.example.fps;

import android.os.Bundle;

import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;

import android.view.MotionEvent;
import android.view.View;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.example.fps.databinding.ActivityMainBinding;

import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration appBarConfiguration;
    private ActivityMainBinding binding;

    private long PERIOD = 1000L;
    private int fpsPrevious = 0;
    private int fpsCurrent = 0;
    private long timePrevious = 0L;
    private long timeCurrent = 0L;
    private long timeDelta  = 0L;
    private long timeThisSecond = 0L;

    /**
     * The main function.
     * @param savedInstanceState
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.toolbar);

        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        appBarConfiguration = new AppBarConfiguration.Builder(navController.getGraph()).build();
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);

        binding.fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });
        Button btn1 = (Button) findViewById(R.id.button_first);
        //Button btn1 = (Button) findViewById(R.id.button_first);
        timeCurrent = System.currentTimeMillis();
        timePrevious = timeCurrent;
        timeCurrent = System.currentTimeMillis();
        timeDelta = timeCurrent - timePrevious;
        btn1.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                switch(event.getAction()) {
                    case MotionEvent.ACTION_MOVE:
                        timePrevious = timeCurrent;
                        timeCurrent = System.currentTimeMillis();
                        timeDelta = timeCurrent - timePrevious;
                        timeThisSecond += timeDelta;
                        if (timeThisSecond >= PERIOD) {
                            fpsPrevious = fpsCurrent;
                            fpsCurrent = 0;
                            timeThisSecond -= PERIOD;
                            btn1.setText(String.valueOf(fpsPrevious));
                            //tvText.text = fpsPrevious.toString();
                        } else {
                            fpsCurrent++;
                        }
                        break;
                    case MotionEvent.ACTION_DOWN:
                        break;
                    case MotionEvent.ACTION_UP:
                    case MotionEvent.ACTION_CANCEL:
                        // RELEASED
                        break;
                }
                return false;
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        return NavigationUI.navigateUp(navController, appBarConfiguration)
                || super.onSupportNavigateUp();
    }
}

File: Android License/Kotlin/FPS/app/src/main/java/com/company/fps/MainActivity.kt
/*
 * Title: MainActivity file.
 * Author: Terrortronics / Bradley Elenbaas (mr.elenbaas@gmail.com)
 * Version: 2
 * Date: September 15, 2023
 *
 * Intellectual Property:
 * Copyright (c) 2023 Bradley Elenbaas. All rights reserved.
 *
 * License:
 * This file is owned by Terrortronics / Bradley Elenbaas.
 * This file observes the Android License.
 */

package com.company.fps

import android.os.Bundle
import android.os.SystemClock
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity


// Original Source: https://www.youtube.com/watch?v=ICiLCEtiwnY&list=PLejN69rU111NiOAX74MS7-nSAIY3uoXYo&index=9&t=209s
/**
 * The main function.
 */
class MainActivity : AppCompatActivity() {

    private lateinit var tvText: TextView

    private var PERIOD = 1000L
    private var fpsPrevious = 0
    private var fpsCurrent = 0
    private var timePrevious = 0L
    private var timeCurrent = 0L
    private var timeDelta  = 0L
    private var timeThisSecond = 0L

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvText = findViewById(R.id.tv_text)
        timeCurrent = SystemClock.elapsedRealtime()
        timePrevious = timeCurrent
        timeCurrent = SystemClock.elapsedRealtime()
        timeDelta = timeCurrent - timePrevious;
        tvText.setOnTouchListener { _, event ->
            timePrevious = timeCurrent;
            timeCurrent = SystemClock.elapsedRealtime();
            timeDelta = timeCurrent - timePrevious;
            timeThisSecond += timeDelta;
            if (timeThisSecond >= PERIOD) {
                fpsPrevious = fpsCurrent;
                fpsCurrent = 0;
                timeThisSecond -= PERIOD;
                tvText.text = fpsPrevious.toString();
            } else {
                fpsCurrent++;
            }
            true
        }
    }
}